build_tools/cmake/test_riscv.sh

```diff
@@ -34,6 +34,7 @@ ctest_args=(
 declare -a label_exclude_args=(
   "^nodocker$"
   "^driver=vulkan$"
+  "^driver=metal$"
   "^driver=cuda$"
```

Comment:
I'd like to eventually remove these label exclusions as part of https://github.com/openxla/iree/issues/12042. Keeping them consistent here for now.

Reply:
(so users not going through these sorts of scripts can run ctest and expect it to only try running tests that actually have a chance of passing - requiring a bunch of labels or a script to get a passing run is confusing)